electron-firmware/src/uavcan/node.cpp

```diff
@@ -60,14 +73,35 @@ void main(unsigned int id, const char* name)
     node.getNodeStatusProvider().setModeOperational();
     node.getNodeStatusProvider().setHealthOk();
 
+    bool timed_out = false;
+    const unsigned int timeout = 30; // seconds
     while (true) {
         node.spin(uavcan::MonotonicDuration::fromMSec(1000 / UAVCAN_SPIN_FREQ));
 
         const auto voltage = 10.f;
 
-        if (data_packet_start_signal_received() && !front_hall_sensor()) {
-            motor_voltage_set(voltage);
-        } else {
+        if (electron_state == INIT && motor_ready()) {
+            electron_state = READY;
+        }
+
+        if (electron_state == READY && data_packet_start_signal_received() && !front_hall_sensor()) {
+            electron_state = RUNNING;
+            start_timer(timeout, &timed_out);
+        }
+
+        if (electron_state == RUNNING) {
+            if (data_packet_start_signal_received() && !front_hall_sensor()) {
+                motor_voltage_set(voltage);
+            } else {
+                motor_voltage_set(0.0f);
+            }
+
+            if (front_hall_sensor() && timed_out) {
```

Comment:
Shouldn't it be `front_hall || timed_out` ?

Reply:
I thought the idea was to continue climbing until timing out, but I'm not sure anymore what we're trying to achieve :thinking:

Reply:
The idea was that it continues to try to climb until the timeout.
I realize now that it should only be `if (timed_out)`.

Reply:
changed now. I added a TIMED_OUT state that should allow to define success only when in ARRIVED state. We could add a message back to the robot for exact score counting.